src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -1213,7 +1211,7 @@ check_session_returned (session_test_t *test, const bson_t *lsid)
       }
    }
 
-   if (!found) {
+   if (!found && ss && ss->last_used_usec != SESSION_NEVER_USED) {
```

Comment:
This makes sense, we explicitly are deciding not to return unused sessions to the pool. Can you add a comment here to that effect?